### PR TITLE
Feature/add discount amount

### PR DESCRIPTION
**Pull request by jacobtani**
* Add DiscountAmount to schema
* Add it to `AllflexProduct`, `OrderedProduct`, `ProductBuilder` and `XmlFormatter`
* Fix specs
* Enable the ordered_product spec and fix errors

### CODE REVIEW

**Comment by jacobtani** [comment]
[Allflex Discount Amount](https://trello.com/c/4kaevTei/2339-allflex-discount-amount)

File: lib/allflex/product_mapper/ordered_product.rb [hunk 1]
```diff
@@ -64,6 +64,10 @@ def expected_price
         (BigDecimal(unit_cost_price) * quantity ).to_digits
       end
 
+      def discount_amount
+        offer_buy_price.nil? ? BigDecimal('0').to_digits : BigDecimal(offer_buy_price).to_digits 
+      end
+
       def unit_cost_price
```
Comment (josephleniston) [inline lib/allflex/product_mapper/ordered_product.rb]: Remove this Method.

File: lib/allflex/product_mapper/ordered_product.rb [hunk 2]
```diff
@@ -64,6 +64,10 @@ def expected_price
         (BigDecimal(unit_cost_price) * quantity ).to_digits
```
Comment (josephleniston) [inline lib/allflex/product_mapper/ordered_product.rb]:
```suggestion
        BigDecimal(offer_price_buy).to_digits 
```

File: lib/allflex/product_mapper/ordered_product.rb [hunk 3]
```diff
@@ -64,6 +64,10 @@ def expected_price
...
       end
 
+      def discount_amount
+        offer_buy_price.nil? ? BigDecimal('0').to_digits : BigDecimal(offer_buy_price).to_digits
```
Comment (josephleniston) [inline lib/allflex/product_mapper/ordered_product.rb]:
```suggestion
        offer_buy_price.nil? ? BigDecimal('0').to_digits : (BigDecimal(offer_price_discount) * -1).to_digits 
```

File: lib/allflex/product_mapper/ordered_product.rb [hunk 4]
```diff
@@ -72,6 +76,14 @@ def unit_cost_price
         ).(@ordered_product)
       end
 
+      def offer_buy_price
```
Comment (josephleniston) [inline lib/allflex/product_mapper/ordered_product.rb]:
```suggestion
      def offer_price_buy
```

File: lib/allflex/product_mapper/ordered_product.rb [hunk 5]
```diff
@@ -72,6 +76,14 @@ def unit_cost_price
...
+        Fn.compose.(
+          Fn.at.("price"),
+          type_extractor.(OFFERED_BUY_PRICE_URN),
```
Comment (josephleniston) [inline lib/allflex/product_mapper/ordered_product.rb]:
```suggestion
          type_extractor.(OFFER_PRICE_BUY_URN),
```

File: lib/allflex/product_mapper/ordered_product.rb [hunk 6]
```diff
@@ -7,7 +7,7 @@ class OrderedProduct
       include Dry::Monads::Result::Mixin
 
       OFFERED_UNIT_COST_PRICE_URN = "urn:lic:offer:price:unit:cost"
-
+      OFFERED_BUY_PRICE_URN = "urn:lic:offer:price:buy"
```
[the remark this answers is not in the document]
Comment (jacobtani) [reply lib/allflex/product_mapper/ordered_product.rb]: Na we still need to use `OFFERED_UNIT_COST_PRICE_URN`, as `urn:lic:offer:price:cost` is never passed in

File: lib/allflex/product_mapper/ordered_product.rb (hunk 1 above)
Comment (jacobtani) [reply lib/allflex/product_mapper/ordered_product.rb]: Still need this.

File: lib/allflex/product_mapper/ordered_product.rb (hunk 6 above)
[the remark this answers is not in the document]
Comment (jacobtani) [reply lib/allflex/product_mapper/ordered_product.rb]: Added in both URNs.

File: lib/allflex/product_mapper/ordered_product.rb (hunk 2 above)
Comment (jacobtani) [reply lib/allflex/product_mapper/ordered_product.rb]: Handling both.

**Pull request merged**